crates/rpcsrv/src/proxy.rs

```diff
@@ -76,11 +77,26 @@ impl<A: ProxyAuthenticator> RpcProxyHandler<A> {
             return Ok((details, conn));
         }
 
-        // Otherwise need to create it.
-        //
-        // TODO: Assumes http, do we want https internally?
-        let url = format!("http://{}:{}", key.ip, key.port);
-        let channel = Endpoint::new(url)?
+        let endpoint = if meta.contains_key("ca_cert_path") && meta.contains_key("domain") {
+            // setup tls
+            let url = format!("https://{}:{}", key.ip, key.port);
+            let ca = std::fs::read_to_string(
+                meta.get("ca_cert_path")
+                    .map(|s| s.to_str())
+                    .transpose()?
+                    .unwrap(),
+            )
+            .unwrap();
+            let tls_conf = ClientTlsConfig::new()
+                .ca_certificate(Certificate::from_pem(ca))
```

Comment:
does this mean we're reading the certificate from the file each time a new request is made?

this feels wrong to me (because we'd just read the same thing again and again) but the truth is that cert-manager can replace/rotate the certificate on a running pod, so maybe this is the correct. Maybe there should be a small ticket or a comment to say "cache this in the service struct, and reread occassionally" (the other model is to use a file system watcher to get notified of the events, but that's worse) 

anyway, I'm just thinking outloud; this seems reasonable and I don't think we should change it now, but a comment might be good.